File: backend/FinTech.Core/Entities/Transaction.cs
using FinTech.Core.Common;
using FinTech.Core.Enums;

namespace FinTech.Core.Entities;

public class Transaction : BaseEntity<Guid>
{
    public string IdempotencyKey { get; set; } = string.Empty;
    public TransactionType Type { get; set; }
    public decimal Amount { get; set; }
    public TransactionStatus Status { get; set; }
    public Guid? LoanId { get; set; }
    public Loan? Loan { get; set; }
    public string Description { get; set; } = string.Empty;
}

File: frontend/src/types/transaction.ts
export type TransactionType = 'Disbursement' | 'Payment' | 'Transfer'
export type TransactionStatus = 'Pending' | 'Completed' | 'Failed'

export interface Transaction {
  id: string
  idempotencyKey: string
  type: TransactionType
  amount: number
  status: TransactionStatus
  loanId?: string
  description: string
  createdAt: string
  updatedAt?: string
}
